File: app/src/main/java/com/natateam/myevents/event/EventPresenerImpl.java
package com.natateam.myevents.event;

import com.natateam.myevents.alarm.AlarmHelper;
import com.natateam.myevents.db.Event;
import com.natateam.myevents.db.RealmHelper;

import javax.inject.Inject;

/**
 * Created by macbook on 30/11/ 15.
 */

public class EventPresenerImpl implements EventContract.EventPresenter {
    EventContract.EventView view;
    RealmHelper realmHelper;
    AlarmHelper alarmHelper;

    @Inject
    public EventPresenerImpl(EventContract.EventView view, RealmHelper realmHelper, AlarmHelper alarmHelper) {
        this.view = view;
        this.realmHelper = realmHelper;
        this.alarmHelper = alarmHelper;
    }




    @Override
    public void isViewAttached() {

    }

    @Override
    public void createOrUpdateEvent(Event event, String title, String desc, String date, String time, long dateMS) {
        realmHelper.createorUpdateEvent(event, title, desc, date, time, dateMS);

    }

    @Override
    public void setData(long id) {
        Event event = realmHelper.getEventById(id);
        view.setData(event);
    }
}

File: app/src/main/java/com/natateam/myevents/event/EventPresenerImpl.kt
package com.natateam.myevents.event

import android.support.v7.app.AppCompatActivity
import com.natateam.myevents.Consts
import com.natateam.myevents.FirebaseHelper
import com.natateam.myevents.alarm.AlarmHelper
import com.natateam.myevents.db.Contact
import com.natateam.myevents.db.Event
import com.natateam.myevents.db.RealmHelper
import com.natateam.myevents.model.ContactModel
import com.natateam.myevents.model.EventModel
import io.realm.RealmObject

import javax.inject.Inject

/**
 * Created by macbook on 30/11/ 15.
 */

class EventPresenerImpl @Inject
constructor(public var view: EventContract.EventView,public  var realmHelper: RealmHelper,public var alarmHelper: AlarmHelper) : EventContract.EventPresenter {
    override fun deleteEvent(id: String) {
        FirebaseHelper.deleteEventById(id)
    }

    override fun deleteContact(id: String) {
        FirebaseHelper.deleteContactById(id)
    }


    override fun createOrUpdateContact(contactModel: ContactModel) {
        FirebaseHelper.addContact(contactModel)
    }



    override fun isViewAttached() {

    }

    override fun createOrUpdateEvent(eventModel: EventModel ) {
        FirebaseHelper.addEvent( eventModel)

    }



    override fun setData(id: String, type: String) {
        if (type == Consts.BIRTH_TYPE){
            FirebaseHelper.getContactyId(id){contactModel -> view.setContactData(contactModel) }
        }else{
            FirebaseHelper.getEventById(id){eventModel -> view.setEventData(eventModel) }
        }
    }

}
